### ISSUE TITLE
Strong variation of accuracy

### ISSUE THREAD

**Issue by AliKalsen** [KieranXWang/HRS#1]
Hi,

I migrated the HRS code from tensorflow 1 to tensorflow 2. Currently, I'm using tensorflow 2.4.2. However, when training the network by calling `train_hrs.py` in the default setting, the accuracies vary from 79% down to 9% every time I run `test_acc.py`.

Nonetheless, when I use the pre-trained weights provided in this repo, I always get about 78% percent accuracy, but tensorflow prints a Data loss error when trying to laod the pre-trained weights. This issue might be caused by the change of the checkpoint files format for saving the weights in tensorflow 2. Actually, this issue doesn't bother me so much when using the pre-trained weights, but somehow I suppose that this might be the issue for the above problem. Unfortunately, I wasn't able to solve this problem.

Did someone have the same issue and knows how to solve this problem? Maybe the authors of the paper have tf 2 migrated code that runs perfectly.
